src/datepicker/calendar-header.js

```diff
@@ -135,66 +168,93 @@ export default class CalendarHeader extends React.Component<HeaderPropsT> {
     };
   }
 
-  setActive = () => {
-    this.props.setActiveState(true);
-  };
-
-  setInactive = () => {
-    this.props.setActiveState(false);
-  };
-
   renderMonthDropdown() {
+    const {date, locale, overrides = {}} = this.props;
+    const [MonthSelect, monthSelectProps] = getOverrides(
+      overrides.MonthSelect,
+      Select,
+    );
+    const selectOverrides = mergeOverrides(
+      this.getSelectOverrides({width: '115px'}),
+      // $FlowFixMe
+      monthSelectProps && monthSelectProps.overrides,
+    );
     const monthOptions = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11].map(M => ({
       id: M,
-      label: getMonthInLocale(M, this.props.locale),
+      label: getMonthInLocale(M, locale),
     }));
     return (
-      <Select
-        onOpen={this.setInactive}
-        onClose={this.setActive}
-        overrides={this.getSelectOverrides({width: '115px'})}
+      <MonthSelect
         clearable={false}
-        value={[{id: getMonth(this.props.date)}]}
+        value={[{id: getMonth(date)}]}
         maxDropdownHeight={'300px'}
-        // $FlowFixMe
-        onChange={this.handleMonthChange}
         options={monthOptions}
         searchable={false}
+        {...monthSelectProps}
+        // Adding event handlers after customers overrides in order to
+        // make sure the components functions as expected
+        // We can extract the handlers from props overrides
+        // and call it along with internal handlers by creating an inline handler
```

Comment:
Thoughts on this approach?

Reply:
Seems reasonable? Are we doing the same thing everywhere else or what rules do we have regarding this?